components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1837,7 +1892,36 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         return [];
     }
 
+    async getWorkspaceEnvVars(
+        ctx: TraceContext,
+        options: GitpodServer.GetWorkspaceEnvVarsOptions,
+    ): Promise<EnvVarWithValue[]> {
+        const user = this.checkUser("getWorkspaceEnvVars");
+        if (options.projectId) {
+            // TODO should we really leak errors to a user? or just return empty and log a warning? block a user?
+            await this.guardProjectOperation(user, options.projectId, "get");
+        }
+
+        const result: EnvVarWithValue[] = [];
+        const { workspaceEnvVars } = await this.resolveWorkspaceEnvVars(user, {
+            repository: undefined, // filter below with access guard
+            projectId: options.projectId,
```

Comment:
Why not workspace.projectId? You have to ensure that anyway, otherwise you're attaching the permission check to user-controlled data? :thinking: